### ISSUE TITLE
XML for download and web site mismatched

### ISSUE THREAD

**Issue by jnothman** [openaustralia/openaustralia#625]
I downloaded the XML from http://data.openaustralia.org/scrapedxml under the assumption that its ids would match those on the web site.

But for example, in http://data.openaustralia.org/scrapedxml/senate_debates/2011-06-15.xml, lords/2011-06-15.4.3 is a speech by Fiona Nash, but the web site shows that speech as [senate/2011-06-15.3.2](http://www.openaustralia.org.au/senate/?gid=2011-06-15.3.2).

What's the story?

If I grab `http://data.openaustralia.org/origxml` and run the OA parser myself am I at least then guaranteed to have a scrapedxml matching www.openaustralia.org?

**Comment by equivalentideas** [comment]
Thanks @jnothman I'm assigning this to @henare cause I think he might be able to provide a more helpful answer that me 😀

**Comment by henare** [comment]
Great question and a valid issue @jnothman :+1: 

>If I grab http://data.openaustralia.org/origxml and run the OA parser myself am I at least then guaranteed to have a scrapedxml matching www.openaustralia.org?

No, I don't think you're guaranteed that.

>What's the story?

GIDs should never change but unfortunately in this project they can. It most commonly happens when Parliament House changes Hansard. They can do this for a few reasons but the most common is fixing mistakes.

When this happens, if we've already (normally automatically) parsed and loaded Hansard, then if we attempt to reparse and reload then GIDs will change if, say, a speech has been inserted. This is mentioned in this wiki page about what you need to do if this happens to avoid resending email alerts:

https://github.com/openaustralia/openaustralia/wiki/Email-alert-tasks

The very likely reason the XML on that date doesn't match what's loaded into OA.org is that in 2014 we had to do a major rewrite of parts of the parser. As part of that process we had to do a [great reparse](https://github.com/openaustralia/openaustralia/issues?utf8=%E2%9C%93&q=Great%20Reparse%20of%202014).

What we didn't do is re_load_ everything then. This means there could be a number of days where the GID in the XML doesn't match what's loaded on the site. If memory serves me correctly we did this because we didn't want a huge amount of broken links from lots of GIDs changing. I guess we thought that was more important at the time than having consistency between the XML and the site?

We should fix this up but I'm not sure of the best approach. It might be worth asking the http://theyworkforyou.com/ team what they do.

This is the first time it's come up so it's probably not going to be a high priority for us to fix it. @jnothman there might be another around doing what you want to do with the data. I'll invite you to our Slack channel to discuss or you can email us contact@oaf.org.au

**Comment by jnothman** [comment]
Yes, having re-parsed the whole collection using the data from  http://data.openaustralia.org/origxml, I can confirm that, for `senate_debates/2011-06-15.xml` at least, origxml and scrapedxml are consistent with the current parser.

I'm basically doing some text mining and visualisation work. But I was hoping to be able to link back to the openaustralia site for users to inspect the automatic analysis / drill down into the data. I see a few options from my perspective:
* don't link
* live with the fact that some links will be broken
* demand (!) that you publish the scrapedxml as loaded on the openaustralia.org web server
* with your help, identify which days have differences, and perhaps provide a mapping from one to the other

**Comment by henare** [comment]
>Yes, having re-parsed the whole collection using the data from http://data.openaustralia.org/origxml, I can confirm that, for senate_debates/2011-06-15.xml at least, origxml and scrapedxml are consistent with the current parser.

Oh nice one, thanks for doing that :+1: 

>I'm basically doing some text mining and visualisation work. But I was hoping to be able to link back to the openaustralia site for users to inspect the automatic analysis / drill down into the data.

:ok_hand: 

>I see a few options from my perspective:
>
>    don't link

:crying_cat_face: 

>live with the fact that some links will be broken

:grimacing: I'm pretty certain that http://theyvoteforyou.org.au/ must have some broken links back to speeches because of this bug :grimacing:

>demand (!) that you publish the scrapedxml as loaded on the openaustralia.org web server

:exclamation: (I actually wonder if we might have the original XML before the great reparse somewhere. I can't see why we would have got rid of it).

>with your help, identify which days have differences, and perhaps provide a mapping from one to the other

This sounds similar to what The Right Solution probably is - get the web application to notice GIDs changing on reload and provide redirects from the old to the new. However I'm sure it's way more complex than that.

I'm definitely happy to keep talking through these options.
